File: client-scalaz/src/main/scala/org/openapitools/client/api/OfferListResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import OfferListResponse._

case class OfferListResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
items: Option[List[OfferResponse]],
hasMoreResults: Option[Boolean],
count: Option[Integer],
start: Option[Integer],
limit: Option[Integer],
countTotal: Option[Long],
returning: Option[String])

object OfferListResponse {
  import DateTimeCodecs._

  implicit val OfferListResponseCodecJson: CodecJson[OfferListResponse] = CodecJson.derive[OfferListResponse]
  implicit val OfferListResponseDecoder: EntityDecoder[OfferListResponse] = jsonOf[OfferListResponse]
  implicit val OfferListResponseEncoder: EntityEncoder[OfferListResponse] = jsonEncoderOf[OfferListResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/offer_list_response.clj
(ns sirqul-io-t-platform.specs.offer-list-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.offer-response :refer :all]
            )
  (:import (java.io File)))


(def offer-list-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :items) (s/coll-of offer-response-spec)
   (ds/opt :hasMoreResults) boolean?
   (ds/opt :count) int?
   (ds/opt :start) int?
   (ds/opt :limit) int?
   (ds/opt :countTotal) int?
   (ds/opt :returning) string?
   })

(def offer-list-response-spec
  (ds/spec
    {:name ::offer-list-response
     :spec offer-list-response-data}))
